### PR TITLE
Add Search Bar Functionality

### PR DESCRIPTION
**Pull request by avanimathur**
- [X] Have you followed the [contributing guidelines](https://github.com/github/opensource.guide/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you explained what your changes do, and why they add value to the Guides?
-----
### Description

The search bar provides users with the opportunity to search for articles without deeply browsing the website. It is especially helpful when the user is looking for precise information and has limited time.
Thus, I have implemented a search bar function.

[screen-capture (1).webm](https://github.com/user-attachments/assets/69d0be5c-6837-4b7b-a457-c36315730e8a)

Please review the changes.
This contribution is under Hacktoberfest'24 .

### CODE REVIEW

File: assets/js/search-index.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     {
       "id": "1",
       "title": "Building Community",
-      "content": "A guide for building a healthy and inclusive open-source community. It covers steps like defining community roles, setting clear expectations, and offering mentorship."
+      "content": "https://opensource.guide/maintaining-balance-for-open-source-maintainers/"
```
Comment (khalilmosavi) [inline assets/js/search-index.json]: assets/js/search-index.json

**Review by khalilmosavi — COMMENTED** [review]
avanimathur:main

**Comment by jmeridth** [comment]
A search capability addition is interesting but the current UI placement (top left above navbar) of the search box is not good. I would suggest adding it to the navbar and also populating results in the main portion of the page or a modal.

Also the dataset to search upon is a static file that would need to be updated with site content. How will it be updated?

This PR feels incomplete IMO.

Do you have additional changes incoming?

I will communicate with other maintainers to see if this implementation/feature is something we're interested in.

**Comment by jmeridth** [comment]
@avanimathur based on lack of response and sparse updates, looking to possibly use deployments for something I'm closing this one. Willing to be proven wrong.

**Pull request closed without merging**